tests/baselines/reference/readonlyTupleAndArrayElaboration.errors.txt

```diff
@@ -55,4 +83,96 @@ tests/cases/compiler/readonlyTupleAndArrayElaboration.ts(24,9): error TS2345: Ar
+    const t3: readonly [1] = [1];
+    const t4: [] = t3;
+          ~~
+!!! error TS4104: The type 'readonly [1]' is 'readonly' and cannot be assigned to the mutable type '[]'.
+    
+    const t5: [1] = [1];
+    const t6: readonly [] = t5;
+          ~~
+!!! error TS2322: Type '[1]' is not assignable to type 'readonly []'.
+!!! error TS2322:   Types of property 'length' are incompatible.
+!!! error TS2322:     Type '1' is not assignable to type '0'.
+    
+    const t7: [1] = [1];
+    const t8: [] = t7;
+          ~~
+!!! error TS2322: Type '[1]' is not assignable to type '[]'.
+!!! error TS2322:   Types of property 'length' are incompatible.
+!!! error TS2322:     Type '1' is not assignable to type '0'.
+    
+    const a1: readonly number[] = [1];
+    const a2: readonly boolean[] = a1;
+          ~~
+!!! error TS2322: Type 'readonly number[]' is not assignable to type 'readonly boolean[]'.
+!!! error TS2322:   Type 'number' is not assignable to type 'boolean'.
+    
+    const a3: readonly number[] = [1];
+    const a4: boolean[] = a3;
+          ~~
+!!! error TS4104: The type 'readonly number[]' is 'readonly' and cannot be assigned to the mutable type 'boolean[]'.
+    
+    const a5: number[] = [1];
+    const a6: readonly boolean [] = a5;
+          ~~
+!!! error TS2322: Type 'number[]' is not assignable to type 'readonly boolean[]'.
+!!! error TS2322:   Type 'number' is not assignable to type 'boolean'.
+    
+    const a7: number[] = [1];
+    const a8: boolean[] = a7;
+          ~~
+!!! error TS2322: Type 'number[]' is not assignable to type 'boolean[]'.
+!!! error TS2322:   Type 'number' is not assignable to type 'boolean'.
+    
+    const ta1: readonly [1] = [1];
+    const ta2: readonly boolean[] = ta1;
+          ~~~
+!!! error TS2322: Type 'readonly [1]' is not assignable to type 'readonly boolean[]'.
+!!! error TS2322:   Type '1' is not assignable to type 'boolean'.
+    
+    const ta3: readonly [1] = [1];
+    const ta4: number[] = ta3;
+          ~~~
+!!! error TS4104: The type 'readonly [1]' is 'readonly' and cannot be assigned to the mutable type 'number[]'.
+    
+    const ta5: [1] = [1];
+    const ta6: readonly boolean[] = ta5;
+          ~~~
+!!! error TS2322: Type '[1]' is not assignable to type 'readonly boolean[]'.
+!!! error TS2322:   Type '1' is not assignable to type 'boolean'.
+    
+    const ta7: [1] = [1];
+    const ta8: boolean[] = ta7;
+          ~~~
+!!! error TS2322: Type '[1]' is not assignable to type 'boolean[]'.
+!!! error TS2322:   Type '1' is not assignable to type 'boolean'.
+    
+    const at1: readonly number[] = [1];
+    const at2: readonly [1] = at1;
+          ~~~
+!!! error TS2741: Property '0' is missing in type 'readonly number[]' but required in type 'readonly [1]'.
+    
+    const at3: readonly number[] = [1];
+    const at4: [1] = at3;
+          ~~~
+!!! error TS2740: Type 'readonly number[]' is missing the following properties from type '[1]': 0, pop, push, reverse, and 4 more.

```

Comment:
How come this one doesn't get the readonly error message?

Reply:
The condition for readonly arrays on the source only checked for arrays on the target - will change and look for tuples.